src/rules/no-empty-string-alt.js

```diff
@@ -0,0 +1,52 @@
+module.exports = {
+  names: ["GH003", "no-empty-string-alt"],
+  description: "Please provide an alternative text for the image.",
+  information: new URL(
+    "https://github.com/github/markdownlint-github/blob/main/docs/rules/GH003-no-empty-string-alt.md",
+  ),
+  tags: ["accessibility", "images"],
+  function: function GH003(params, onError) {
+    const htmlTagsWithImages = params.parsers.markdownit.tokens.filter(
+      (token) => {
+        return (
+          (token.type === "html_block" && token.content.includes("<img")) ||
+          (token.type === "inline" &&
+            token.content.includes("<img") &&
+            token.children.some((child) => child.type === "html_inline"))
+        );
+      },
+    );
+    const inlineImages = params.parsers.markdownit.tokens.filter(
+      (token) =>
+        token.type === "inline" &&
+        token.children.some((child) => child.type === "image"),
+    );
+
+    const htmlAltRegex = new RegExp(/alt=""|alt=''/, "gid");
```

Comment:
nit, for readability:

```suggestion
    const htmlAltRegex = /alt=['"]['"]/gid;
```